Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM rust:1.60.0 as rustbuilder
+WORKDIR /tools
+RUN git clone https://github.com/parallaxsecond/parsec-tool.git
+RUN cd parsec-tool && cargo build
+WORKDIR /app
+COPY rust/parsec-hello-decrypt parsec-hello-decrypt
+RUN cd parsec-hello-decrypt && cargo build
+
+FROM golang:1.18.1 as gobuilder
+WORKDIR /app
+COPY go/parsec-hello-decrypt parsec-hello-decrypt
+RUN cd parsec-hello-decrypt && go get parsec/parsec-hello-decrypt && go build .
+
+FROM ubuntu:20.04
+WORKDIR /tools
+COPY --from=rustbuilder /tools/parsec-tool/target/debug/parsec-tool .
+WORKDIR /app/rust
+COPY --from=rustbuilder /app/parsec-hello-decrypt/target/debug/parsec-hello-decrypt .
+WORKDIR /app/go
+COPY --from=gobuilder /app/parsec-hello-decrypt/parsec-hello-decrypt .
+WORKDIR /demo
+COPY --from=rustbuilder /tools/parsec-tool/tests/parsec-cli-tests.sh .
```

Comment:
What's the reason for this to be copied across?

Reply:
That's a multi-layer image build. We don't need to have all the build environments in the final image. So, we build tools in separate layers and only copy what we need to the result layer

Reply:
Yes, what I meant was - I don't see it being used anywhere, so why do we copy it in?

Reply:
If the question was about `parsec-cli-tests.sh` - I had a previous conversation with @anta5010 and we decided that it would be useful to have the CLI test script available in the container for optional execution, so I copied it in, but there's no option to execute it yet. We can do that later, but I figured I might as well copy it in because that was easy.